kotest-assertions/kotest-assertions-json/src/jvmMain/kotlin/io/kotest/assertions/json/JsonMatchers.kt

```diff
@@ -1,81 +1,208 @@
 package io.kotest.assertions.json
 
-import com.fasterxml.jackson.databind.ObjectMapper
-import com.fasterxml.jackson.module.kotlin.registerKotlinModule
 import com.jayway.jsonpath.JsonPath
 import com.jayway.jsonpath.PathNotFoundException
+import io.kotest.assertions.failure
 import io.kotest.matchers.Matcher
 import io.kotest.matchers.MatcherResult
 import io.kotest.matchers.should
 import io.kotest.matchers.shouldNot
+import kotlin.contracts.ExperimentalContracts
+import kotlin.contracts.contract
+import kotlin.reflect.typeOf
+
+@OptIn(ExperimentalContracts::class)
+infix fun Json?.shouldMatchJson(json: Json?) {
+    contract {
+        returns() implies (this@shouldMatchJson != null)
+    }
+
+    this should matchJson(json)
+}
+
+@OptIn(ExperimentalContracts::class)
+infix fun Json?.shouldNotMatchJson(json: Json?) {
+    contract {
+        returns() implies (this@shouldNotMatchJson != null)
+    }
+
+    this shouldNot matchJson(json)
+}
```

Comment:
I don't think this contract is correct.

If I execute
```
null shouldNotMatchJson null
```

It seems that `null == null` will work, and thus both values will be null. This isn't what the contract states

Reply:
Good catch, fixed in a365ed5b.